controller/src/main/java/io/pravega/controller/store/stream/tables/serializers/StreamConfigurationRecordSerializer.java

```diff
@@ -0,0 +1,165 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.controller.store.stream.tables.serializers;
+
+import com.google.common.base.Preconditions;
+import io.pravega.client.stream.RetentionPolicy;
+import io.pravega.client.stream.ScalingPolicy;
+import io.pravega.client.stream.StreamConfiguration;
+import io.pravega.common.io.serialization.RevisionDataInput;
+import io.pravega.common.io.serialization.RevisionDataOutput;
+import io.pravega.common.io.serialization.VersionedSerializer;
+import io.pravega.controller.store.stream.tables.StreamConfigurationRecord;
+
+import java.io.IOException;
+
+import static io.pravega.controller.store.stream.tables.StreamConfigurationRecord.RetentionPolicyRecord;
+import static io.pravega.controller.store.stream.tables.StreamConfigurationRecord.RetentionPolicyRecord.RetentionPolicyRecordBuilder;
+import static io.pravega.controller.store.stream.tables.StreamConfigurationRecord.ScalingPolicyRecord;
+import static io.pravega.controller.store.stream.tables.StreamConfigurationRecord.ScalingPolicyRecord.ScalingPolicyRecordBuilder;
+import static io.pravega.controller.store.stream.tables.StreamConfigurationRecord.StreamConfigurationRecordBuilder;
+
+public class StreamConfigurationRecordSerializer
+        extends VersionedSerializer.WithBuilder<StreamConfigurationRecord,
+        StreamConfigurationRecordBuilder> {
+    @Override
+    protected byte getWriteVersion() {
+        return 0;
+    }
+
+    @Override
+    protected void declareVersions() {
+        version(0).revision(0, this::write00, this::read00);
+    }
+
+    private void read00(RevisionDataInput revisionDataInput,
+                        StreamConfigurationRecordBuilder streamConfigurationRecordBuilder)
+            throws IOException {
+        // streamConfigurationRecordBuilder.streamConfiguration()
+        StreamConfiguration.StreamConfigurationBuilder streamConfigurationBuilder = StreamConfiguration.builder();
+        streamConfigurationBuilder.scope(revisionDataInput.readUTF())
+                .streamName(revisionDataInput.readUTF())
+                .scalingPolicy(ScalingPolicyRecord.SERIALIZER.deserialize(revisionDataInput).getScalingPolicy())
+                .retentionPolicy(RetentionPolicyRecord.SERIALIZER.deserialize(revisionDataInput).getRetentionPolicy());
+        streamConfigurationRecordBuilder.streamConfiguration(streamConfigurationBuilder.build())
+                .updating(revisionDataInput.readBoolean());
+    }
+
+    private void write00(StreamConfigurationRecord streamConfigurationRecord, RevisionDataOutput revisionDataOutput)
+            throws IOException {
+        Preconditions.checkNotNull(streamConfigurationRecord);
```

Comment:
consider using the `protected void beforeSerialization(TargetType object)` method defined in the `VersionedSerializer` class. That has a blank implementation by default, but it allows you to do precondition checks before serialization. This:
1. Helps your code be more organized. That method's contract defines what kind of exception you should throw if the object is not ready to be serialized, so that it is properly handled by the serializer.
2. Helps you in the case you will be streaming objects out to a non-rewindable stream (this is not the case here though, but it's a good practice anyway).